### PR TITLE
01 create book ent

### PR DESCRIPTION
**Pull request by anastasiia-fi**

### CODE REVIEW

File: src/main/java/config/AppConfig.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package config;
+
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@ComponentScan(basePackages = "java")
+public class AppConfig {
```
Comment (liliia-ponomarenko) [inline src/main/java/config/AppConfig.java:8]: This class can be removed

File: pom.xml [hunk 2]
```diff
@@ -28,6 +28,21 @@
...
+        <dependency>
+            <groupId>com.mysql</groupId>
+            <artifactId>mysql-connector-j</artifactId>
+            <scope>runtime</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.hibernate</groupId>
+            <artifactId>hibernate-core</artifactId>
+            <version>6.2.7.Final</version>
+        </dependency>
```
Comment (liliia-ponomarenko) [inline pom.xml:45]: Do you really  need this dependency

File: src/main/java/dao/BookRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't save book to DB", e);
```
Comment (liliia-ponomarenko) [inline src/main/java/dao/BookRepositoryImpl.java:26]: Better use your custom EntityNotFoundException here

File: src/main/java/model/Book.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+
+@Entity
+@Table(name = "books")
+public class Book {
+
```
Comment (liliia-ponomarenko) [inline src/main/java/model/Book.java:14]:
```suggestion
```

File: src/main/java/model/Book.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    @Column(nullable = false, unique = true)
+    private String isbn;
+    @Column(nullable = false)
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
+
+    public Book() {
+
+    }
```
Comment (liliia-ponomarenko) [inline src/main/java/model/Book.java:31]: Do you really need this constructor?

File: src/test/resources/application.properties [hunk 6]
```diff
@@ -0,0 +1,6 @@
+spring.datasource.url=jdbc:h2:mem:testdb
+spring.datasource.driverClassName=org.h2.Driver
+spring.datasource.username=sa
+spring.datasource.password=password
+spring.jpa.database-platform=org.hibernate.dialect.H2Dialect
+
```
Comment (liliia-ponomarenko) [inline src/test/resources/application.properties:6]:
```suggestion
```

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s fix a few mistakes ;)

File: src/main/java/dao/BookRepositoryImpl.java (hunk 3 above)
Comment (anastasiia-fi) [reply src/main/java/dao/BookRepositoryImpl.java:26]: I create it in the next task (we have it in description of our hw) so I left here DataProcessingException

File: src/test/resources/application.properties (hunk 6 above)
Comment (liliia-ponomarenko) [reply src/test/resources/application.properties:6]: not fixed(

**Review by liliia-ponomarenko — APPROVED** [review]
Good job! Let’s try to merge ;)

**Pull request merged**